modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2683,10 +2694,26 @@ def broadcast_apply_full_axis(
             kw["dtypes"] = pandas.Series(
                 [np.dtype(dtypes)] * len(kw["columns"]), index=kw["columns"]
             )
+
+        if not keep_partitioning:
+            if kw["row_lengths"] is None:
+                if new_index is not None and axis == 0:
+                    kw["row_lengths"] = get_length_list(
+                        axis_len=len(new_index), num_splits=new_partitions.shape[0]
+                    )
+                elif axis == 1:
+                    kw["row_lengths"] = self._row_lengths_cache
+            if kw["column_widths"] is None:
+                if new_columns is not None and axis == 1:
+                    kw["column_widths"] = get_length_list(
+                        axis_len=len(new_columns), num_splits=new_partitions.shape[1]
+                    )
+                elif axis == 0:
+                    kw["column_widths"] = self._column_widths_cache
```

Comment:
```suggestion
            if kw["row_lengths"] is None and new_index is not None:
                kw["row_lengths"] = get_length_list(
                    axis_len=len(new_index), num_splits=new_partitions.shape[0]
                )
            if kw["column_widths"] is None and new_columns is not None:
                kw["column_widths"] = get_length_list(
                    axis_len=len(new_columns), num_splits=new_partitions.shape[1]
                )
```
the resulting shape may change for any axis, we can't use the old row/column lengths

Reply:
btw, maybe we can move this logic into `.__make_init_labels_args`, it seems like it does extra work for post-processing the `kw`, so it should be better to put the logic in there so no post-processing wouldn't be needed

Reply:
You are right, however your changes will fail the repartition tests.